homeassistant/components/nobo_hub/climate.py

```diff
@@ -0,0 +1,406 @@
+            _LOGGER.warning(
+                "Can not turn off (or on) any zone, because week profile '%s' was not found",
+                command_off_name,
+            )
+        else:
+            _LOGGER.debug(
+                "To turn off any heater, week profile %s '%s' will be used",
+                command_off_id,
+                command_off_name,
+            )
+
+            # Find ON command (week profile) for the different zones:
+            command_on_dict = config_entry.options.get(CONF_COMMAND_ON)
+            if command_on_dict is None or command_on_dict.keys().__len__ == 0:
+                _LOGGER.warning(
+                    "Not possible to turn on any zone, because ON week profile was not specified"
+                )
+            else:
+                _set_on_commands(command_on_by_id, command_on_dict, hub)
+
+    # Add zones as entities
+    async_add_devices(
+        NoboZone(zone_id, hub, command_off_id, command_on_by_id.get(zone_id))
+        for zone_id in hub.zones
+    )
+
+    hass.bus.async_listen_once(EVENT_HOMEASSISTANT_STOP, hub.stop)
+
+    return True
+
+
+def _set_on_commands(command_on_by_id, command_on_dict, hub):
+    for zone_id, zone in hub.zones.items():
+        # Replace unicode non-breaking space (used in Nobø Ecohub) with space
+        zone_name = zone[ATTR_NAME].replace("\xa0", " ")
+        if zone_name in command_on_dict:
+            command_on_name = command_on_dict[zone_name]
+            if command_on_name is None or command_on_name == "":
+                _LOGGER.warning(
+                    "Can not turn on (or off) zone '%s', because ON week profile was not specified",
+                    zone_name,
+                )
+            else:
+                command_on_id = _get_id_from_name(command_on_name, hub.week_profiles)
+                if command_on_id is None or command_on_id == "":
+                    _LOGGER.warning(
+                        "Can not turn on (or off) zone '%s', because ON week profile '%s' was not found",
+                        zone_name,
+                        command_on_name,
+                    )
+                else:
+                    _LOGGER.debug(
+                        "To turn on heater %s '%s', week profile %s '%s' will be used",
+                        zone_id,
+                        zone_name,
+                        command_on_id,
+                        command_on_name,
+                    )
+                    command_on_by_id[zone_id] = command_on_id
+
+
+def _get_id_from_name(name, dictionary):
+    for key in dictionary.keys():
+        # Replace unicode non-breaking space (used in Nobø Ecohub) with space
+        if dictionary[key][ATTR_NAME].replace("\xa0", " ") == name:
+            return key
+    return None
+
+
+class NoboZone(ClimateEntity):
+    """Representation of a Nobø zone. A Nobø zone consists of a group of physical devices that are controlled as a unity."""
+
```

Comment:
```suggestion
class NoboZone(ClimateEntity):
    """Representation of a Nobø zone. 
    
    A Nobø zone consists of a group of physical devices that are
    controlled as a unity.
    """
    
    _attr_max_temp = MAX_TEMPERATURE
    _attr_min_temp = MIN_TEMPERATURE
    _attr_precision = PRECISION_TENTHS
    _attr_preset_modes = PRESET_MODES
    _attr_should_poll = False
    _attr_supported_features = SUPPORT_FLAGS
    _attr_temperature_unit = TEMP_CELSIUS
    

```

See: <https://developers.home-assistant.io/docs/core/entity#entity-class-attributes>